### ISSUE TITLE
`type_info` is not exported correctly

### ISSUE THREAD

**Issue by huangqinjin** [microsoft/STL#4643]
Run preprocessor on `std.ixx` and inspect its output:
```shell
>cl /EHsc /std:c++latest /P "%VCToolsInstallDir%\modules\std.ixx"
Microsoft (R) C/C++ Optimizing Compiler Version 19.40.33808 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

/std:c++latest is provided as a preview of language features from the latest C++
working draft, and we're eager to hear about bugs and suggestions for improvements.
However, note that these features are provided as-is without support, and subject
to changes or removal as the working draft evolves. See
https://go.microsoft.com/fwlink/?linkid=2045807 for details.

std.ixx
```
Search `type_info` in `std.i` from top, we see that the first statement is in `eh.h`
```c++
extern "C++" { // attach declarations to the global module, see N4910 [module.unit]/7
    class type_info;
} // extern "C++"
```
and later in `vcruntime_typeinfo.h` it is exported:
```c++
extern "C++" { 
export class type_info {};
}
```

I believe this violates https://eel.is/c++draft/module.interface#6
> A redeclaration of an entity X is implicitly exported if X was introduced by an exported declaration; otherwise it shall not be exported[.](https://eel.is/c++draft/module.interface#6.sentence-1)

**Comment by StephanTLavavej** [comment]
Thanks - I forgot to search VCRuntime for forward-declaring its own machinery. (This would be easier if MSVC enforced the Standard rule; I've got an active bug requesting that from the compiler team.)

**Comment by StephanTLavavej** [comment]
Created internal MSVC-PR-549242.

**Comment by StephanTLavavej** [comment]
Fix merged for VS 2022 17.11 Preview 3. Thanks again for the report! :heart_eyes_cat:
